x-pack/legacy/plugins/graph/public/components/guidance_panel/guidance_panel.tsx

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { ReactNode } from 'react';
+import { EuiPanel, EuiFlexGroup, EuiFlexItem, EuiIcon, EuiText, EuiLink } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import classNames from 'classnames';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+export interface GuidancePanelProps {
+  onFillWorkspace: () => void;
+  onOpenFieldPicker: () => void;
+  onOpenDatasourcePicker: () => void;
+  hasDatasource: boolean;
+  hasFields: boolean;
+}
+
+function ListItem({
+  children,
+  state,
+}: {
+  state: 'done' | 'active' | 'disabled';
+  children: ReactNode;
+}) {
+  return (
+    <li
+      className={classNames('gphGuidancePanel__item', {
+        'gphGuidancePanel__item--disabled': state === 'disabled',
+      })}
+      aria-disabled={state === 'disabled'}
+    >
+      {state !== 'disabled' && (
+        <span
+          className={classNames('gphGuidancePanel__itemIcon', {
+            'gphGuidancePanel__itemIcon--done': state === 'done',
+          })}
+          aria-hidden={true}
+        >
+          <EuiIcon type={state === 'active' ? 'sortRight' : 'check'} />
+        </span>
+      )}
+      {children}
```

Comment:
Just to add to your review soup, wrap this in `<EuiText/>` to fix your text line heights.